src/instruments/src/EFB/Ground/Pages/ServicesPage.tsx

```diff
@@ -38,159 +43,524 @@ const ServiceButtonWrapper: FC<ServiceButtonWrapperProps> = ({ children, classNa
+            // as the useEffect will never be called.
+            if (doorOpenState === 1) {
+                setTimeout(() => {
+                    dispatch(setButtonState(ServiceButtonState.ACTIVE));
+                }, 5000);
             }
-        }
-    }, [jetWayActive, cargoActive, cateringActive, fuelingActive, powerActive, pushBack, wheelChocksEnabled, conesEnabled]);
-
-    const handleClick = (callBack: () => void, event: React.MouseEvent, gameSync?, disabledButton?: string) => {
-        if (!tugActive) {
-            if (!activeButtons.map((b: StatefulButton) => b.id).includes(event.currentTarget.id)) {
-                dispatch(addActiveButton({ id: event.currentTarget.id, state: STATE_WAITING, callBack, value: gameSync }));
-                if (disabledButton) {
-                    dispatch(addDisabledButton(disabledButton));
+        } else if (serviceButtonStateRef.current === ServiceButtonState.CALLED) {
+            // When in state CALLED another click on the button cancels the request.
+            // This prevents another click after a "called" has been cancelled
+            // to avoid state getting out of sync.
+            dispatch(setButtonState(ServiceButtonState.DISABLED));
+            setTimeout(() => {
+                dispatch(setButtonState(ServiceButtonState.INACTIVE));
+            }, 5500);
+        } else {
+            console.assert(serviceButtonStateRef.current === ServiceButtonState.ACTIVE,
+                'Expected %s to be in state %s but was in state %s',
+                ServiceButton[serviceButton],
+                ServiceButtonState[ServiceButtonState.ACTIVE],
+                ServiceButtonState[serviceButtonStateRef.current]);
+            dispatch(setButtonState(ServiceButtonState.RELEASED));
+            // If there is no service vehicle/jet-bridge available the door would
+            // never receive a close event, so we need to set the button state
+            // to inactive after a timeout.
+            setTimeout(() => {
+                if (doorOpenState === 1) {
+                    dispatch(setButtonState(ServiceButtonState.INACTIVE));
                 }
-                callBack();
-            } else {
-                const index = activeButtons.map((b: StatefulButton) => b.id).indexOf(event.currentTarget.id);
+            }, 5000);
+        }
 
-                if (index > -1) {
-                    dispatch(removeActiveButton(index));
-                }
-                if (disabledButton) {
-                    const disabledIndex = disabledButtons.indexOf(disabledButton);
-                    dispatch(removeDisabledButton(disabledIndex));
+        // Door Button: enable door button after a timeout if it was disabled
+        if (doorButtonState === ServiceButtonState.DISABLED) {
+            setTimeout(() => {
+                // service button could have been pressed again in the meantime
+                if (serviceButtonStateRef.current < ServiceButtonState.CALLED) {
```

Comment:
I would avoid doing these types of comparisons with an enum as it isn't very intuitive, is error-prone if the order of enum members is ever changed, and not very explicit. I would change this to more explicitly define the condition for the behavior specified.

Reply:
Why would anybody every change the order of an enum in a language where the the order mattes? 
This is a very common practice in C/C++ to have serial/ordered enums used like this by taking advantage of the fact that enums are in fact just automatically assigned incremental const integers. With this it is much easier to write these if-statements otherwise I would have to always compare all states I want to cover in the if. It is less code and I think it is readable if you are used to it. 
I think in a language where you use .map or closures so often using enums like this is surely one of the more readable parts :D. Just look how complicated the previous code for this page was - unreadable and unmaintainable. 
But lets discuss on Discord :)